packages/vm/src/runCall.ts

```diff
@@ -74,14 +78,23 @@ export interface RunCallOpts {
 /**
  * @ignore
  */
-export default function runCall(this: VM, opts: RunCallOpts): Promise<EVMResult> {
+export default async function runCall(this: VM, opts: RunCallOpts): Promise<EVMResult> {
   const block = opts.block ?? Block.fromBlockData({}, { common: this._common })
 
   const txContext = new TxContext(
     opts.gasPrice ?? BigInt(0),
     opts.origin ?? opts.caller ?? Address.zero()
   )
 
+  if (opts.skipBalance) {
+    const caller = opts.caller ?? Address.zero()
```

Comment:
i think instead of defining caller twice, once here and once in `new Message`, you can define above the if statement `const caller` then in `new Message` just do `{ caller }`. then we are using one "source of truth"